staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go

```diff
@@ -51,3 +53,33 @@ func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) (*hand
 
 	return openAPIVersionedService, spec
 }
+
+func (oa OpenAPI) InstallV3(c *restful.Container, mux *mux.PathRecorderMux) *handler3.OpenAPIService {
+	openAPIVersionedService, err := handler3.NewOpenAPIService(nil)
+	if err != nil {
+		klog.Fatalf("Failed to create OpenAPIService: %v", err)
+	}
+
+	err = openAPIVersionedService.RegisterOpenAPIV3VersionedService("/openapi/v3", mux)
+	if err != nil {
+		klog.Fatalf("Failed to register versioned open api spec for root: %v", err)
+	}
+
+	grouped := make(map[string][]*restful.WebService)
+
+	for _, t := range c.RegisteredWebServices() {
+		// Strip the "/" prefix from the name
+		gvName := t.RootPath()[1:]
+		grouped[gvName] = []*restful.WebService{t}
+	}
+
+	for gv, ws := range grouped {
+		spec, err := builder3.BuildOpenAPISpec(ws, oa.Config)
+		if err != nil {
+			klog.Errorf("Failed to build OpenAPI v3 for group %s, %q", gv, err)
+
+		}
+		openAPIVersionedService.UpdateGroupVersion(gv, spec)
```

Comment:
which GVs does grouped include here? Are these supposed to be the static groups in kube-apiserver? Does this mean we have to call this func late in the startup? What about dynamic APIServices?

Reply:
Yes, this is the list of all static GVs in kube-apiserver. Both this function and InstallV2 are called in `PrepareRun` of the apiserver.

Dynamic APIServices are added in the aggregator (same place as for openapi v2)